contracts/FeralfileArtworkV3_3.sol

```diff
@@ -0,0 +1,140 @@
+    using Strings for uint256;
+    using Counters for Counters.Counter;
+
+    address public minter;
+    Counters.Counter private tokenCounter;
+    mapping(uint256 => string) private artworkCIDs; // artworkID => artwork cids
+
+    modifier onlyMinter() {
+        require(_msgSender() == minter, "Only minter can call this function");
+        _;
+    }
+
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        uint256 secondarySaleRoyaltyBPS_,
+        address royaltyPayoutAddress_,
+        string memory contractURI_,
+        string memory tokenBaseURI_,
+        bool isBurnable_,
+        bool isBridgeable_,
+        address minter_
+    )
+        FeralfileExhibitionV3_2(
+            name_,
+            symbol_,
+            secondarySaleRoyaltyBPS_,
+            royaltyPayoutAddress_,
+            contractURI_,
+            tokenBaseURI_,
+            isBurnable_,
+            isBridgeable_
+        )
+    {
+        minter = minter_;
+    }
+
+    /// @notice tokenURI A distinct Uniform Resource Identifier (URI) for a given asset.
+    /// @param tokenId - the id of token
+    function tokenURI(
+        uint256 tokenId
+    ) public view virtual override returns (string memory) {
+        require(
+            _exists(tokenId),
+            "ERC721Metadata: URI query for nonexistent token"
+        );
+
+        ArtworkEditionIndex memory aei = allArtworkEditionsIndex[tokenId];
+
+        require(
+            bytes(artworkCIDs[aei.artworkID]).length != 0,
+            "Artwork CID does not exist"
+        );
+
+        string memory baseURI = _tokenBaseURI;
+        if (bytes(baseURI).length == 0) {
+            baseURI = "ipfs://";
+        }
+
+        return
+            string(
+                abi.encodePacked(
+                    baseURI,
+                    artworkCIDs[aei.artworkID],
+                    "/",
+                    tokenId.toString()
+                )
+            );
+    }
+
+    /// @notice getArtworkIDByTokenID use for getting artwork ID from token ID
+    /// @param _tokenId - the id of edition
+    function getArtworkIDByTokenID(
+        uint256 _tokenId
+    ) public view returns (uint256) {
+        return allArtworkEditionsIndex[_tokenId].artworkID;
+    }
+
+    /// @notice updateArtworkCIDs use for update CIDs for artworks
+    /// @param _artworkIDs - array of artwork IDs
+    /// @param _artworkCIDs - array of ipfs CIDs
+    function updateArtworkCIDs(
+        uint256[] memory _artworkIDs,
+        string[] memory _artworkCIDs
+    ) public onlyAuthorized {
+        require(_artworkIDs.length == _artworkCIDs.length, "Invalid input");
+        for (uint256 i = 0; i < _artworkIDs.length; i++) {
+            require(
+                bytes(artworks[_artworkIDs[i]].title).length != 0,
+                "Artwork does not exist"
+            );
+            artworkCIDs[_artworkIDs[i]] = _artworkCIDs[i];
+        }
+    }
+
+    function batchMint(
```

Comment:
// I think we need docs even this is a dummy function. By saying why we need this.

Reply:
Added comments